spec/domain/etl/ga/views_and_navigation_service_spec.rb

```diff
@@ -68,5 +68,43 @@
           .to yield_successive_args(arg1, arg2)
       end
     end
+
+    context 'when report data has very long page_path' do
+      before do
+        allow(google_client).to receive(:fetch_all) do
+          [
+            build_report_data(
+              build_report_row(dimensions: [junk_page_path], metrics: %w(1 1 1 1 1 1 1 1))
+            ),
+          ]
+        end
+      end
+
+      it 'yields shortened page_path value' do
+        arg = [
+          a_hash_including(
+            'page_path' => junk_page_path.slice(1500),
+            'pviews' => 1,
+            'upviews' => 1,
+            'entrances' => 1,
+            'exits' => 1,
+            'bounces' => 1,
+            'page_time' => 1,
+            'date' => '2018-02-20',
+          )
+        ]
+
+        expect { |probe| subject.find_in_batches(date: date, batch_size: 1, &probe) }
+          .to yield_with_args(arg)
+      end
+    end
+  end
+
+private
+
+  def junk_page_path
```

Comment:
What about generating the string with ruby?

```ruby
str = "/".concat("a" * 1600)
```

Reply:
That's definitely nicer... [fixed](https://github.com/alphagov/content-performance-manager/commit/cc0c35e8add2d9086b41d0b57b018c967951d20d)